> which one? I'm sure I tried `dotnet-sdk_6`, but I didn't do all feature levels.

Literally all of them:
```
these 9 derivations will be built:
  /nix/store/qw4rxwbngl0jwm4ddddl92q0ghpj5168-dotnet-sdk-wrapped-6.0.136-test-console-single-file-fs.drv
  /nix/store/2x20a7ykk087rk7xv199sghfly8sxpz6-dotnet-sdk-wrapped-6.0.136-test-console-single-file-fs-run.drv
  /nix/store/3zq879mgb66py0f62zxn8di76ldvf9qd-dotnet-sdk-wrapped-6.0.136-test-console-run-fs.drv
  /nix/store/67iididxjrvn6y4ln8lh1lqk9kkv1wr3-dotnet-sdk-wrapped-6.0.136-test-console-publish-fs.drv
  /nix/store/hlr5w1ba3k2qvkf64f48bi5sc38is4zs-dotnet-sdk-wrapped-6.0.136-test-console-self-contained-fs.drv
  /nix/store/cwp1j36dh2zlf8ib0kc96j4a96jli9m9-dotnet-sdk-wrapped-6.0.136-test-console-self-contained-fs-run.drv
  /nix/store/hxipn2cqcgj7x83gmjcqzb291zi7fh9p-dotnet-sdk-wrapped-6.0.136-test-web-fs.drv
  /nix/store/lpr7y3nzv09r7vjy5bflx70kwsqbdhzf-dotnet-sdk-wrapped-6.0.136-test-web-fs-run.drv
  /nix/store/yxhw02mqjxysz38gmfdml16jbl2hg1q9-dotnet-sdk-wrapped-6.0.136-test-console-publish-fs-run.drv
building '/nix/store/3zq879mgb66py0f62zxn8di76ldvf9qd-dotnet-sdk-wrapped-6.0.136-test-console-run-fs.drv'...
building '/nix/store/67iididxjrvn6y4ln8lh1lqk9kkv1wr3-dotnet-sdk-wrapped-6.0.136-test-console-publish-fs.drv'...
building '/nix/store/hlr5w1ba3k2qvkf64f48bi5sc38is4zs-dotnet-sdk-wrapped-6.0.136-test-console-self-contained-fs.drv'...
building '/nix/store/qw4rxwbngl0jwm4ddddl92q0ghpj5168-dotnet-sdk-wrapped-6.0.136-test-console-single-file-fs.drv'...
building '/nix/store/hxipn2cqcgj7x83gmjcqzb291zi7fh9p-dotnet-sdk-wrapped-6.0.136-test-web-fs.drv'...
error: builder for '/nix/store/67iididxjrvn6y4ln8lh1lqk9kkv1wr3-dotnet-sdk-wrapped-6.0.136-test-console-publish-fs.drv' failed with exit code 1;
       last 18 log lines:
       > Running phase: unpackPhase
       > The template "Console App" was created successfully.
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configureNuget
       > The template "NuGet Config" was created successfully.
       >
       > Processing post-creation actions...
       >
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > Microsoft (R) Build Engine version 17.0.4+85d71072f for .NET
       > Copyright (C) Microsoft Corporation. All rights reserved.
       >
       >   Determining projects to restore...
       > /build/test/test.fsproj : error NU1101: Unable to find package FSharp.Core. No packages exist with this id in source(s): _nix
       >   Failed to restore /build/test/test.fsproj (in 77 ms).
       For full logs, run 'nix log /nix/store/67iididxjrvn6y4ln8lh1lqk9kkv1wr3-dotnet-sdk-wrapped-6.0.136-test-console-publish-fs.drv'.
error: builder for '/nix/store/hxipn2cqcgj7x83gmjcqzb291zi7fh9p-dotnet-sdk-wrapped-6.0.136-test-web-fs.drv' failed with exit code 1;
       last 18 log lines:
       > Running phase: unpackPhase
       > The template "ASP.NET Core Empty" was created successfully.
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configureNuget
       > The template "NuGet Config" was created successfully.
       >
       > Processing post-creation actions...
       >
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > Microsoft (R) Build Engine version 17.0.4+85d71072f for .NET
       > Copyright (C) Microsoft Corporation. All rights reserved.
       >
       >   Determining projects to restore...
       > /build/test/test.fsproj : error NU1101: Unable to find package FSharp.Core. No packages exist with this id in source(s): _nix
       >   Failed to restore /build/test/test.fsproj (in 75 ms).
       For full logs, run 'nix log /nix/store/hxipn2cqcgj7x83gmjcqzb291zi7fh9p-dotnet-sdk-wrapped-6.0.136-test-web-fs.drv'.
error: builder for '/nix/store/hlr5w1ba3k2qvkf64f48bi5sc38is4zs-dotnet-sdk-wrapped-6.0.136-test-console-self-contained-fs.drv' failed with exit code 1;
       last 18 log lines:
       > Running phase: unpackPhase
       > The template "Console App" was created successfully.
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configureNuget
       > The template "NuGet Config" was created successfully.
       >
       > Processing post-creation actions...
       >
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > Microsoft (R) Build Engine version 17.0.4+85d71072f for .NET
       > Copyright (C) Microsoft Corporation. All rights reserved.
       >
       >   Determining projects to restore...
       > /build/test/test.fsproj : error NU1101: Unable to find package FSharp.Core. No packages exist with this id in source(s): _nix
       >   Failed to restore /build/test/test.fsproj (in 96 ms).
       For full logs, run 'nix log /nix/store/hlr5w1ba3k2qvkf64f48bi5sc38is4zs-dotnet-sdk-wrapped-6.0.136-test-console-self-contained-fs.drv'.
error: builder for '/nix/store/qw4rxwbngl0jwm4ddddl92q0ghpj5168-dotnet-sdk-wrapped-6.0.136-test-console-single-file-fs.drv' failed with exit code 1;
       last 18 log lines:
       > Running phase: unpackPhase
       > The template "Console App" was created successfully.
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configureNuget
       > The template "NuGet Config" was created successfully.
       >
       > Processing post-creation actions...
       >
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > Microsoft (R) Build Engine version 17.0.4+85d71072f for .NET
       > Copyright (C) Microsoft Corporation. All rights reserved.
       >
       >   Determining projects to restore...
       > /build/test/test.fsproj : error NU1101: Unable to find package FSharp.Core. No packages exist with this id in source(s): _nix
       >   Failed to restore /build/test/test.fsproj (in 96 ms).
       For full logs, run 'nix log /nix/store/qw4rxwbngl0jwm4ddddl92q0ghpj5168-dotnet-sdk-wrapped-6.0.136-test-console-single-file-fs.drv'.
error: builder for '/nix/store/3zq879mgb66py0f62zxn8di76ldvf9qd-dotnet-sdk-wrapped-6.0.136-test-console-run-fs.drv' failed with exit code 1;
       last 14 log lines:
       > Running phase: unpackPhase
       > The template "Console App" was created successfully.
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configureNuget
       > The template "NuGet Config" was created successfully.
       >
       > Processing post-creation actions...
       >
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       >
       > The build failed. Fix the build errors and run again.
       For full logs, run 'nix log /nix/store/3zq879mgb66py0f62zxn8di76ldvf9qd-dotnet-sdk-wrapped-6.0.136-test-console-run-fs.drv'.
error: 1 dependencies of derivation '/nix/store/cwp1j36dh2zlf8ib0kc96j4a96jli9m9-dotnet-sdk-wrapped-6.0.136-test-console-self-contained-fs-run.drv' failed to build
error: 1 dependencies of derivation '/nix/store/2x20a7ykk087rk7xv199sghfly8sxpz6-dotnet-sdk-wrapped-6.0.136-test-console-single-file-fs-run.drv' failed to build
error: 1 dependencies of derivation '/nix/store/lpr7y3nzv09r7vjy5bflx70kwsqbdhzf-dotnet-sdk-wrapped-6.0.136-test-web-fs-run.drv' failed to build
error: 1 dependencies of derivation '/nix/store/yxhw02mqjxysz38gmfdml16jbl2hg1q9-dotnet-sdk-wrapped-6.0.136-test-console-publish-fs-run.drv' failed to build
error: build of '/nix/store/2x20a7ykk087rk7xv199sghfly8sxpz6-dotnet-sdk-wrapped-6.0.136-test-console-single-file-fs-run.drv', '/nix/store/3zq879mgb66py0f62zxn8di76ldvf9qd-dotnet-sdk-wrapped-6.0.136-test-console-run-fs.drv', '/nix/store/cwp1j36dh2zlf8ib0kc96j4a96jli9m9-dotnet-sdk-wrapped-6.0.136-test-console-self-contained-fs-run.drv', '/nix/store/lpr7y3nzv09r7vjy5bflx70kwsqbdhzf-dotnet-sdk-wrapped-6.0.136-test-web-fs-run.drv', '/nix/store/yxhw02mqjxysz38gmfdml16jbl2hg1q9-dotnet-sdk-wrapped-6.0.136-test-console-publish-fs-run.drv' failed
```

--- a/pkgs/development/compilers/dotnet/wrapper.nix
+++ b/pkgs/development/compilers/dotnet/wrapper.nix
@@ -104,7 +104,7 @@ stdenvNoCC.mkDerivation (finalAttrs: {
           let
             sdk = finalAttrs.finalPackage;
             built = stdenv.mkDerivation {
-              name = "dotnet-test-${name}";
+              name = "${sdk.name}-test-${name}";
               buildInputs = [ sdk ] ++ buildInputs ++ lib.optional (usePackageSource) sdk.packages;
               # make sure ICU works in a sandbox
               propagatedSandboxProfile = toString sdk.__propagatedSandboxProfile;

--- a/pkgs/development/compilers/dotnet/wrapper.nix
+++ b/pkgs/development/compilers/dotnet/wrapper.nix
@@ -288,6 +288,7 @@ stdenvNoCC.mkDerivation (finalAttrs: {
           # yes, older SDKs omit the comma
           cs = mkConsoleTests "C#" "cs" "Hello,?\\ World!";
           fs = mkConsoleTests "F#" "fs" "Hello\\ from\\ F#";
+          vb = mkConsoleTests "VB" "vb" "Hello,?\\ World!";
         };
 
         web = lib.recurseIntoAttrs {

--- a/pkgs/development/compilers/dotnet/wrapper.nix
+++ b/pkgs/development/compilers/dotnet/wrapper.nix
@@ -82,6 +82,7 @@ stdenvNoCC.mkDerivation (finalAttrs: {
   postFixup = lib.optionalString (unwrapped ? man) ''
     ln -s ${unwrapped.man} "$man"
   '';
+
   passthru = unwrapped.passthru // {
     inherit unwrapped;
     tests =
@@ -91,6 +92,7 @@ stdenvNoCC.mkDerivation (finalAttrs: {
             name,
             stdenv ? stdenvNoCC,
             template,
+            lang ? null,
             usePackageSource ? false,
             build,
             buildInputs ? [ ],
@@ -106,16 +108,28 @@ stdenvNoCC.mkDerivation (finalAttrs: {
               buildInputs = [ sdk ] ++ buildInputs ++ lib.optional (usePackageSource) sdk.packages;
               # make sure ICU works in a sandbox
               propagatedSandboxProfile = toString sdk.__propagatedSandboxProfile;
-              unpackPhase = ''
-                mkdir test
-                cd test
-                dotnet new ${template} -o . --no-restore
-              '';
+              unpackPhase =
+                let
+                  unpackArgs =
+                    [ template ]
+                    ++ lib.optionals (lang != null) [
+                      "-lang"
+                      lang
+                    ];
+                in
+                ''
+                  mkdir test
+                  cd test
+                  dotnet new ${lib.escapeShellArgs unpackArgs} -o . --no-restore
+                '';
               buildPhase = build;
               dontPatchELF = true;
             };
           in
-          if run == null then
+          # older SDKs don't include an embedded FSharp.Core package
+          if lang == "F#" && lib.versionOlder sdk.version "6.0.400" then
+            null
+          else if run == null then
             built
           else
             runCommand "${built.name}-run"
@@ -142,61 +156,90 @@ stdenvNoCC.mkDerivation (finalAttrs: {
                 + run
               );
 
-        # Setting LANG to something other than 'C' forces the runtime to search
-        # for ICU, which will be required in most user environments.
-        checkConsoleOutput = command: ''
-          output="$(LANG=C.UTF-8 ${command})"
-          # yes, older SDKs omit the comma
-          [[ "$output" =~ Hello,?\ World! ]] && touch "$out"
-        '';
-      in
-      unwrapped.passthru.tests or { }
-      // {
-        version = testers.testVersion (
-          {
-            package = finalAttrs.finalPackage;
+        mkConsoleTests =
+          lang: suffix: output:
+          let
+            # Setting LANG to something other than 'C' forces the runtime to search
+            # for ICU, which will be required in most user environments.
+            checkConsoleOutput = command: ''
+              output="$(LANG=C.UTF-8 ${command})"
+              [[ "$output" =~ ${output} ]] && touch "$out"
+            '';
+
+            mkConsoleTest =
+              { name, ... }@args:
+              mkDotnetTest (
+                args
+                // {
+                  name = "console-${name}-${suffix}";
+                  template = "console";
+                  inherit lang;
+                }
+              );
+          in
+          lib.recurseIntoAttrs {
+            run = mkConsoleTest {
+              name = "run";
+              build = checkConsoleOutput "dotnet run";
+            };
+
+            publish = mkConsoleTest {
+              name = "publish";
+              build = "dotnet publish -o $out/bin";
+              run = checkConsoleOutput "$src/bin/test";
+            };
+
+            self-contained = mkConsoleTest {
+              name = "self-contained";
+              usePackageSource = true;
+              build = "dotnet publish --use-current-runtime --sc -o $out";
+              runtime = null;
+              run = checkConsoleOutput "$src/test";
+            };
+
+            single-file = mkConsoleTest {
+              name = "single-file";
+              usePackageSource = true;
+              build = "dotnet publish --use-current-runtime -p:PublishSingleFile=true -o $out/bin";
+              runtime = null;
+              run = checkConsoleOutput "$src/bin/test";
+            };
           }
-          // lib.optionalAttrs (type != "sdk") {
-            command = "dotnet --info";
-          }
-        );
-      }
-      // lib.optionalAttrs (type == "sdk") (
-        {
-          console = mkDotnetTest {
-            name = "console";
-            template = "console";
-            build = checkConsoleOutput "dotnet run";
+          // lib.optionalAttrs finalAttrs.finalPackage.hasILCompiler {
+            aot = mkConsoleTest {
+              name = "aot";
+              stdenv = if stdenv.hostPlatform.isDarwin then swiftPackages.stdenv else stdenv;
+              usePackageSource = true;
+              buildInputs =
+                [
+                  zlib
+                ]
+                ++ lib.optional stdenv.hostPlatform.isDarwin (
+                  with darwin;
+                  with apple_sdk.frameworks;
+                  [
+                    swiftPackages.swift
+                    Foundation
+                    CryptoKit
+                    GSS
+                    ICU
+                  ]
+                );
+              build = ''
+                dotnet restore -p:PublishAot=true
+                dotnet publish -p:PublishAot=true -o $out/bin
+              '';
+              runtime = null;
+              run = checkConsoleOutput "$src/bin/test";
+            };
           };
 
-          publish = mkDotnetTest {
-            name = "publish";
-            template = "console";
-            build = "dotnet publish -o $out/bin";
-            run = checkConsoleOutput "$src/bin/test";
-          };
-
-          self-contained = mkDotnetTest {
-            name = "self-contained";
-            template = "console";
-            usePackageSource = true;
-            build = "dotnet publish --use-current-runtime --sc -o $out";
-            runtime = null;
-            run = checkConsoleOutput "$src/test";
-          };
-
-          single-file = mkDotnetTest {
-            name = "single-file";
-            template = "console";
-            usePackageSource = true;
-            build = "dotnet publish --use-current-runtime -p:PublishSingleFile=true -o $out/bin";
-            runtime = null;
-            run = checkConsoleOutput "$src/bin/test";
-          };
-
-          web = mkDotnetTest {
-            name = "web";
+        mkWebTest =
+          lang: suffix:
+          mkDotnetTest {
+            name = "web-${suffix}";
             template = "web";
+            inherit lang;
             build = "dotnet publish -o $out/bin";
             runtime = finalAttrs.finalPackage.aspnetcore;
             runInputs = [
@@ -228,36 +271,29 @@ stdenvNoCC.mkDerivation (finalAttrs: {
             '';
             runAllowNetworking = true;
           };
-        }
-        // lib.optionalAttrs finalAttrs.finalPackage.hasILCompiler {
-          aot = mkDotnetTest {
-            name = "aot";
-            stdenv = if stdenv.hostPlatform.isDarwin then swiftPackages.stdenv else stdenv;
-            template = "console";
-            usePackageSource = true;
-            buildInputs =
-              [
-                zlib
-              ]
-              ++ lib.optional stdenv.hostPlatform.isDarwin (
-                with darwin;
-                with apple_sdk.frameworks;
-                [
-                  swiftPackages.swift
-                  Foundation
-                  CryptoKit
-                  GSS
-                  ICU
-                ]
-              );
-            build = ''
-              dotnet restore -p:PublishAot=true
-              dotnet publish -p:PublishAot=true -o $out/bin
-            '';
-            runtime = null;
-            run = checkConsoleOutput "$src/bin/test";
-          };
-        }
-      );
+      in
+      unwrapped.passthru.tests or { }
+      // {
+        version = testers.testVersion (
+          {
+            package = finalAttrs.finalPackage;
+          }
+          // lib.optionalAttrs (type != "sdk") {
+            command = "dotnet --info";
+          }
+        );
+      }
+      // lib.optionalAttrs (type == "sdk") ({
+        console = lib.recurseIntoAttrs {
+          # yes, older SDKs omit the comma
+          cs = mkConsoleTests "C#" "cs" "Hello,?\\ World!";
+          fs = mkConsoleTests "F#" "fs" "Hello\\ from\\ F#";
+        };
+
+        web = lib.recurseIntoAttrs {
+          cs = mkWebTest "C#" "cs";
+          fs = mkWebTest "F#" "fs";
+        };
+      });
   };
 })

